apps/openglcompute/test_oglc_avg.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+const int CHANNELS = 4;
+
+void blur(std::string suffix, ImageParam input) {
+    input.set_bounds(2, 0, CHANNELS).set_stride(0, CHANNELS).set_stride(2, 1);
+
+    Var x("x"), y("y"), c("c");
+
+    Func clamped("clamped");
+    clamped(x, y, c) = input(clamp(x, input.left(), input.right()),
+                             clamp(y, input.top(), input.bottom()), c);
+
+
+    Func blur_x("blur_x");
+    blur_x(x, y, c) = (clamped(x, y, c) +
+        clamped(x + 1, y, c) +
```

Comment:
This blur shifts the image by 1 pixel. For consistency (we use this example a lot in Halide, e.g. halide-lang.org), I think this should use a kernel starting at -1 instead of 0.